pkg/config/mesh/mesh.go

```diff
@@ -40,63 +40,144 @@ var (
 
 // DefaultProxyConfig for individual proxies
 func DefaultProxyConfig() meshconfig.ProxyConfig {
+	if TestMode {
+		return meshconfig.ProxyConfig{
+			ConfigPath:             constants.ConfigPathDir,
+			BinaryPath:             constants.BinaryPathFilename,
+			ServiceCluster:         constants.ServiceClusterName,
+			DrainDuration:          types.DurationProto(45 * time.Second),
+			ParentShutdownDuration: types.DurationProto(60 * time.Second),
+			DiscoveryAddress:       constants.DiscoveryPlainAddress,
+			StatsdUdpAddress:       "",
+			EnvoyMetricsService:    &meshconfig.RemoteService{Address: ""},
+			EnvoyAccessLogService:  &meshconfig.RemoteService{Address: ""},
+			ProxyAdminPort:         15000,
+			ControlPlaneAuthPolicy: meshconfig.AuthenticationPolicy_NONE,
+			CustomConfigFile:       "",
+			Concurrency:            0,
+			StatNameLength:         189,
+			Tracing:                nil,
+			StatusPort:             15020,
+		}
+	}
+	// TODO: include revision based on REVISION env
+	// TODO: set default namespace based on POD_NAMESPACE env
 	return meshconfig.ProxyConfig{
+		// missing: ConnectTimeout: 10 * time.Second,
 		ConfigPath:             constants.ConfigPathDir,
-		BinaryPath:             constants.BinaryPathFilename,
 		ServiceCluster:         constants.ServiceClusterName,
 		DrainDuration:          types.DurationProto(45 * time.Second),
 		ParentShutdownDuration: types.DurationProto(60 * time.Second),
-		DiscoveryAddress:       constants.DiscoveryPlainAddress,
-		StatsdUdpAddress:       "",
-		EnvoyMetricsService:    &meshconfig.RemoteService{Address: ""},
-		EnvoyAccessLogService:  &meshconfig.RemoteService{Address: ""},
 		ProxyAdminPort:         15000,
+		Concurrency:            2,
 		ControlPlaneAuthPolicy: meshconfig.AuthenticationPolicy_NONE,
-		CustomConfigFile:       "",
-		Concurrency:            0,
-		StatNameLength:         189,
-		Tracing:                nil,
-		StatusPort:             15020,
+		DiscoveryAddress:       "istiod.istio-system.svc:15012",
```

Comment:
This value should be a const, it is widely used

Reply:
Well, see my TODO... It will soon be 
"istiod-" + os.Getenv(revision) + "." + os.Getenv(POD_NAMESPACE).